### DISCUSSION TITLE
sync/history returns Response 500

### DISCUSSION THREAD

**Discussion by rohitsathish** [trakt/trakt-api#349] [Q&A] [answered]
Hi, I am trying to update my watch history by sending a POST request to https://api.trakt.tv/sync/history on python.

Here's my code thats giving a Response 500 - 

```
test_value = {'movies': [{'title': '7500',
   'year': 2019,
   'ids': {'trakt': 357152,
    'slug': '7500-2019',
    'imdb': 'tt6436726',
    'tmdb': 509585},
   'watched_at': '2021-06-05T00%3A00%3A00.000Z'}],
 'seasons': [],
 'episodes': [],
 'shows': []}

headers = {
    'Content-type': 'application/json',
    'trakt-api-key': 'xxxx',
    'trakt-api-version': '2'
}

import requests as req
url = 'https://api.trakt.tv/sync/history'
req.post(url, data = test_value, headers = headers)
```

Am I doing something wrong or is this a server side issue? 

Thanks

**Comment by tysonkerridge** [comment, marked as answer]
Unsure if it's the cause of the 500 but it looks like you might be missing your authentication header:
eg. `'Authorization': 'Bearer {access_token}'`
And perhaps your `watched_at` date is incorrectly formatted:
eg. `'watched_at': '2014-09-01T09:10:11.000Z'`

**Reply by rohitsathish** [reply]
> Unsure if it's the cause of the 500 but it looks like you might be missing your authentication header: eg. `'Authorization': 'Bearer {access_token}'` And perhaps your `watched_at` date is incorrectly formatted: eg. `'watched_at': '2014-09-01T09:10:11.000Z'`

I tried it with these changes but I'm still getting the 500 Response. Is the same request working on your end? Then maybe I'm doing something wrong.

**Reply by rectifyer** [reply]
Is it possible to get the raw request and response, including headers? I'm curious what the python is actually creating and sending.

**Reply by tysonkerridge** [reply]
Double check all your response headers, chances are it'll have more information for you. I tested quickly in Postman using your headers/body and get a 500 without the `Authorization` header as I pointed out. I got a 201 when it's included. You'll most likely need to check why the Authorisation header isn't being sent. 

Error headers returned to me for reference 
```
www-authenticate: Bearer realm="Doorkeeper", error="invalid_token", error_description="The access token is invalid"
x-error-message: Internal server error. Please open a support issue and include your full API request and response, including all headers.
```
